### PR TITLE
DOC: improve numpy.histogram2d() documentation

### PR DESCRIPTION
**Pull request by marysia**
Clarify necessity of H.T in code example. Resolves issue #18929

Co-authored-by: Laura Kopf <60775505+lkopf@users.noreply.github.com>

### CODE REVIEW

**Review by mattip — APPROVED** [review]
Looks good to me.

**Review by rgommers — APPROVED** [review]
LGTM too, thanks @marysia and Laura!

File: numpy/lib/twodim_base.py [hunk 1]
```diff
@@ -705,7 +705,9 @@ def histogram2d(x, y, bins=10, range=None, normed=None, weights=None,
     >>> x = np.random.normal(2, 1, 100)
     >>> y = np.random.normal(1, 1, 100)
     >>> H, xedges, yedges = np.histogram2d(x, y, bins=(xedges, yedges))
-    >>> H = H.T  # Let each row list bins with common y range.
+    >>> # Histogram does not follow Cartesian convention (see Notes),
+    >>> # therefore transpose H for visualization purposes.
+    >>> H = H.T
```
Comment (rossbar) [inline numpy/lib/twodim_base.py:710]: Just a very minor suggestion, but you can pull the explanatory text outside of executable prompts to make it stand out a little more:

```suggestion

    Histogram does not follow Cartesian convention (see Notes),
    therefore transpose H for visualization purposes.
    
    >>> H = H.T
```

One thing to be aware for this is that there must be blank lines between the executable prompts and the text.

**Review by rossbar — APPROVED** [review]
Of course, the changes are also fine as-is!

**Comment by mattip** [comment]
The [rendered doc](https://19723-908607-gh.circle-artifacts.com/0/doc/build/html/reference/generated/numpy.histogram2d.html?highlight=histogram2d#numpy.histogram2d)

**Comment by mattip** [comment]
Thanks @marysia and @lkopf

**Comment by afkowalski** [comment]
Thank you very much for doing this.

**Comment by lkopf** [comment]
I don't appear as an official co-author. My username is @lkopf

**Pull request merged**